activesupport/lib/active_support/core_ext/date_time/calculations.rb

```diff
@@ -138,6 +138,12 @@ def utc_offset
 
   # Layers additional behavior on DateTime#<=> so that Time and ActiveSupport::TimeWithZone instances can be compared with a DateTime
   def <=>(other)
-    super other.kind_of?(Infinity) ? other : other.to_datetime
+    if other.kind_of?(Infinity)
+      super
+    elsif other.respond_to? :to_datetime
+      super other.to_datetime
+    else
```

Comment:
@egilburg I think here the goal is to explicitly point out the return value. In this case `nil`.